packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -415,6 +472,62 @@ class _ModalBottomSheetRoute<T> extends PopupRoute<T> {
   }
 }
 
+/// A curve that progresses linearly until a specified [startingPoint], at which
```

Comment:
We can't fundamentally redefine what Curve means in a subclass. The [base class](https://api.flutter.dev/flutter/animation/Curve-class.html) says:
```
A curve must map t=0.0 to 0.0 and t=1.0 to 1.0.
```

You could subclass Curve2D instead.

Reply:
I think that condition is satisfied in ```transform```

Reply:
First statement in transform is:
```
if (t < startingPoint) {
  return t;
}
```